libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -750,9 +750,11 @@ public async Task<ConversationsResult> GetConversationsAsync(string serviceUrl,
                 throw new ArgumentNullException(nameof(credentials));
             }
 
-            var connectorClient = CreateConnectorClient(serviceUrl, credentials);
-            var results = await connectorClient.Conversations.GetConversationsAsync(continuationToken, cancellationToken).ConfigureAwait(false);
-            return results;
+            using (var connectorClient = CreateConnectorClient(serviceUrl, credentials))
```

Comment:
Will this now be calling dispose on the ServerClient's HttpClient?

Reply:
Yup. `ServiceClient`, the parent class of `ConnectorClient` has a `disposeHttpClient` optional constructor parameter which defaults to `true`. When `true`, when `Dispose()` gets called on the `ConnectorClient`, the passed-in `HttpClient` will be disposed automatically. Check the AutoRest code [here ](https://github.com/SyntaxC4-MSFT/AutoRest/blob/master/Microsoft.Rest/ClientRuntime/ServiceClient.cs#L170)if you're curious.

Reply:
This was discussed in the PR review and we think Eric's point is a genuine concern. We discussed other approaches to go around the problem. We should discuss more.

Reply:
Notes from discussion: The fact  that the `ServiceClient` will dispose the http client is not good, since it would dispose custom http clients and trigger ObjectDisposedExceptions.  Investigating with Eric.

Reply:
Added a new ctor that allows httpclient disposal. Fortunately we can capture the calls to connector client in our codebase that have a custom http client, and move them to the new constructor. Also adding tests for the disposing or not disposing depending on the parameters.